internal/job/currency-job.go

```diff
@@ -2,17 +2,16 @@ package job
 
 import (
 	"log"
-	"time"
 
-	"genesis-currency-api/internal/service"
+	"genesis-currency-api/pkg/interface/service"
+
 	"github.com/robfig/cron/v3"
 )
 
 // UpdateCurrencyJob is a cron function to update currency service cache.
 // It is executed every hour.
-func UpdateCurrencyJob(currencyService *service.CurrencyService) {
-	scheduler := cron.New(cron.WithLocation(time.UTC))
-	_, err := scheduler.AddFunc("0 * * * *", func() {
+func UpdateCurrencyJob(cron *cron.Cron, currencyService service.CurrencyService) {
```

Comment:
As for me service.CurrencyService is not very good name for interface. And also, as @AndrewSergienko was mentioned https://github.com/GenesisEducationKyiv/software-engineering-school-4-0-AlwaysSayNo/pull/3#discussion_r1643902395

Interfaces is  better to locate near it is used.

Something like this

```go
type CurrencyUpdater interface {
	UpdateCurrencyRates() error
}

// UpdateCurrencyJob is a cron function to update currency service cache.
// It is executed every hour.
func UpdateCurrencyJob(cron *cron.Cron, currencyService CurrencyUpdater) {
	_, err := cron.AddFunc("0 * * * *", func() {
		log.Println("Start job: Update Currency Rates")

		if err := currencyService.UpdateCurrencyRates(); err != nil {
			log.Printf("error with: Update Currency Rates - %v\n", err)
		} else {
			log.Println("Finish job: Update Currency Rates")
		}
	})
	if err != nil {
		log.Printf("failed to start UpdateCurrencyJob scheduler: %v\n", err)
		return
	}
}
```

You just need only one method `UpdateCurrencyRates` not all methods from interface

Reply:
Done.